### ISSUE TITLE
Add test suite

### ISSUE THREAD

**Issue by kernelmethod** [The-Encryption-Compendium/TECv2#14]
Some kind of test suite should be set up to ensure that the site works correctly before it's deployed.
- Since the site is static right now, it might be worth focusing on functional tests for the time being.
- That said, Vue.js provides a [unit testing framework](https://vuejs.org/v2/guide/unit-testing.html) that can also be used to test the site.

**Comment by dkg** [comment]
I'd recommend including in any sort of testing plan unit tests like:

- verification that the python code is all explicitly and correctly typed to the extent possible (see #33)
- liveness checks for all entries that have URLs (see #39)
- bibtex consistency check (see #46)

**Comment by kernelmethod** [comment]
I agree, I think all of those would be good to add to the site. I've just added type checking (albeit only for `generate_compendium.py`, so far) in #48; I'll need to put some more thought into the other two ideas.
